src/main/java/org/jenkinsci/plugins/registry/notification/events/impl/ManifestDeleted.java

```diff
@@ -0,0 +1,43 @@
+package org.jenkinsci.plugins.registry.notification.events.impl;
+
+import hudson.Extension;
+import jenkins.model.Jenkins;
+import net.sf.json.JSONObject;
+import org.jenkinsci.plugins.registry.notification.Messages;
+import org.jenkinsci.plugins.registry.notification.events.EventType;
+import org.jenkinsci.plugins.registry.notification.events.EventTypeDescriptor;
+import org.kohsuke.stapler.DataBoundConstructor;
+
+import javax.annotation.CheckForNull;
+
+public class ManifestDeleted extends EventType {
+
+    public static final String JSON_TYPE = "MANIFEST_DELETE";
+
+    @DataBoundConstructor
+    public ManifestDeleted() {}
+
+    @Override
+    public String getType() { return JSON_TYPE; }
+
+    @Override
+    public String getTimeStamp(JSONObject contents) { return contents.optString("deletedAt"); }
```

Comment:
`EventType.buildEnvironment` doesn't seem to handle `null` getting returned here.

Reply:
Fixing